### PR TITLE
Psmithcrl/feature detection

### PR DESCRIPTION
**Pull request by psmithcrl**

### CODE REVIEW

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh [hunk 1]
```diff
@@ -3846,6 +3893,134 @@ class MULTISENSE_API ApriltagParams {
...
+        /**
+         * numberOfFeatures
+         * The maximum features to be searched for in one image.
+         */
+        uint32_t m_numberOfFeatures;
+
+        /**
+         * grouping
+         * Enable/Disable the grouping feature in feaure detection.
+         * TODO: improve description
```
Comment (mattalvarado) [inline source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:3939]: Can we address this TODO?

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh [hunk 2]
```diff
@@ -3846,6 +3893,134 @@ class MULTISENSE_API ApriltagParams {
...
+         */
+        bool m_grouping;
+
+        /**
+         * motion
+         * Enable / disable motion detection in the image.
+         * TODO: improve description
```
Comment (mattalvarado) [inline source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:3946]: Can we address this TODO?

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh [hunk 3]
```diff
@@ -3846,6 +3893,134 @@ class MULTISENSE_API ApriltagParams {
...
+         * Current recommended settings.
+         * Full    Resolution: 5000 Features @5FPS
+         * Quarter Resolution: 1500 Features @15FPS
+         *
+         * @param numberOfFeatures The maximum number of features.
+         */
+
+        void setNumberOfFeatures(const uint32_t &numberOfFeatures)    {
+
+            if (numberOfFeatures > 5000)
```
Comment (mattalvarado) [inline source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:3983]: Should we define this limit with the other constants in `MultiSenseTypes.hh`?

File: source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh (hunk 1 above)
Comment (psmithcrl) [reply source/LibMultiSense/include/MultiSense/MultiSenseTypes.hh:3939]: Yes, will do

**Review by mattalvarado — APPROVED** [review]
Looks good to me!

**Pull request merged**